src/main/java/jewellery/inventory/service/ProductService.java

```diff
@@ -198,20 +200,16 @@ private void disassembleProductContent(Product product) {
     }
   }
 
-  private void moveResourceInProductToResourceInUser(Product product) {
+  private void moveQuantityFromResourcesInProductToResourcesInUser(Product product) {
     List<ResourceInProduct> resourcesInProduct = product.getResourcesContent();
-    User owner = product.getOwner();
-    moveQuantityFromResourcesInProductToResourcesInUser(resourcesInProduct, owner);
-  }
-
-  private void moveQuantityFromResourcesInProductToResourcesInUser(
-      List<ResourceInProduct> resourcesInProduct, User owner) {
     resourcesInProduct.forEach(
         resourceInProduct -> {
           resourceInUserService.addResourceToUserNoLog(
-              getResourceInUserRequest(owner, resourceInProduct));
+              getResourceInUserRequest(product.getOwner(), resourceInProduct));
           resourceInProduct.setQuantity(0);
```

Comment:
How about instead of resourceInProduct.setQuantity(0), we just do `resourceInProductRepository.delete(resourceInProduct.getId())`? In this case we don't need to `clearResourceInProductRepository` that fetches all of the records in ResourceInProduct table just to remove the ones that are with 0 quantity.